openfasoc/generators/gdsfactory-gen/process_input.py

```diff
@@ -0,0 +1,233 @@
+		if savepath.parent.is_dir():
+			with savepath.open("w") as savefile:
+				if savecode:
+					savefile.write(self.code.get_code())
+				else:
+					savefile.writelines([r+"\n" for r in self.conversation_responses])
+		else:
+			raise FileNotFoundError("the directory you specfied does not exist")
+	
+	def print_to_stream(self, toprint: Optional[str]=None):
+		if toprint is None:
+			toprint=""
+		self.outputstream.write(toprint+"\n")
+	
+
+	def read_from_stream(self) -> str:
+		response = self.inputstream.readline()
+		#self.conversation_responses.append(response)
+		return response
+	
+	def __save_response(self, response: str):
+		self.conversation_responses.append(response.strip())
+
+	def process_next_input(self, text_input: str) -> bool:
+		"""main driver for doing things
+		returns True if session is ongoing
+		"""
+		self.backup = self.__backup()
+		sentences = nltk.sent_tokenize(text_input)
+		for sentence in sentences:
+			saveresponse = True
+			sentence = sentence.strip().removesuffix(".")
+			words = nltk.word_tokenize(sentence)
+			mode_indicator = words[0].strip().replace("-","").lower()
+			# parse user input
+			if mode_indicator[0]=="h":# help
+				saveresponse = False
+				syntaxes = list()
+				syntaxes.append("place an genidhere called/named compnamehere with paramshere")#place_syntax
+				syntaxes.append("route between/from port1 and/to port2 using routetype with paramshere")#route_syntax
+				syntaxes.append("move compname [by] (x,y)")# absolute move syntax
+				syntaxes.append("move compname [filler words] direction [filler words] reference_comp [by separation]")# relative move syntax
+				import_syntax = "import comp1, comp2 from mod1, and comp3 from some/path/mod.py"
+				import_syntax += "\n\tNOTE: imports from glayout only need component name"
+				import_syntax += "\n\tNOTE: if mod name is not specified, it is assumed to be the component name"
+				syntaxes.append(import_syntax) # import
+				syntaxes.append("create/define [a/an] param_type parameter called/named paramname")# parameter
+				syntaxes.append("create/define [a/an] var_type variable called/named varname =/equal valorexpr")#variable
+				syntaxes.append("save/dump [conversation/code] to pathtosaveto\n\tNOTE: save code is assumed if neither conversation nor code are specified")#
+				# print all
+				self.print_to_stream("\nBelow are valid sentences:")
+				#for i, syntax in enumerate(syntaxes):
+				for syntax in syntaxes:
+					#self.print_to_stream(str(i)+" : "+syntax)
+					self.print_to_stream(syntax)
+				self.print_to_stream()
+			elif mode_indicator[0]=="i":# import
+				imports = sentence.replace("and",",").replace("import","").replace("from","").strip().split(",")
+				for modimport in imports:
+					if modimport=="" or modimport.isspace():
+						continue
+					words = modimport.strip().split()
+					compname = words[0]
+					modpath = words[1] if len(words)>1 else None
+					aliases = [compname]#TODO implement comp aliasing
```

Comment:
Has this todo been fixed?

Reply:
no, this is an idea we can do down the line.